src/shared/auth/platform/guards/company-or-professional-auth.guard.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { Request } from 'express';
+import { Reflector } from '@nestjs/core';
+import { PostgresPrismaService } from '@/config/prisma/postgres.services';
+import { PinoLogger } from 'nestjs-pino';
+import { IS_PUBLIC_KEY } from '@/shared/decorators/isPublic.decorator';
+
+@Injectable()
+export class CompanyOrProfessionalAuthGuard implements CanActivate {
+  private readonly logger = new PinoLogger({
+    renameContext: CompanyOrProfessionalAuthGuard.name,
+  });
+
+  constructor(
+    private jwtService: JwtService,
+    private reflector: Reflector,
+    private prisma: PostgresPrismaService,
+  ) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const isPublic = this.reflector.getAllAndOverride<boolean>(IS_PUBLIC_KEY, [
+      context.getHandler(),
+      context.getClass(),
+    ]);
+    if (isPublic) {
+      return true;
+    }
+
+    const request = context.switchToHttp().getRequest<Request>();
+    const token = this.extractTokenFromHeader(request);
+
+    if (!token) {
+      throw new UnauthorizedException('No token provided');
+    }
+
+    // Try Company Auth
+    const company = await this.authenticate(token, request, 'company');
+    if (company) {
+      return true;
+    }
+
+    // Try Professional Auth
+    const user = await this.authenticate(token, request, 'user');
+    if (user) {
+      return true;
+    }
+
+    throw new UnauthorizedException('Invalid token');
+  }
```

Comment:
you don't have to do that, as basically

all u need is to check the prefix of the client id
`ck` for company and `pk` for professional, passed on that you take the action